content/blogposts/2024/local-code-completion-with-ollama-and-cody.md

```diff
@@ -0,0 +1,135 @@
+
+```bash
+ollama pull codellama:7b
+```
+
+**Important Note**: Your ability to run a model is heavily dependent on your hardware. The larger the model, the more resources you will need to succesfully run it.
+
+### Running LLMs locally
+
+Once you have downloaded a model, you can run it locally by specifying the model name. For example, to run the `codellama` model, you would run the following command:
+
+```bash
+ollama run codellama
+```
+
+By default Ollama will run the model directly in your terminal. Once the model is running, you can interact with it by typing in your prompt and pressing enter. For example, once the terminal is running, you can type in the following prompt:
+
+```bash
+Write a JavaScript function that takes a string and returns the number of vowels in the string.
+```
+
+And press enter. Ollama will respond with an output like this::
+
+![Ollama terminal output](https://storage.googleapis.com/sourcegraph-assets/blog/local-code-completion-with-ollama-and-cody/ollama-terminal.png)
+
+### Ollama REST API
+
+Ollama also provides a REST API that you can use to interact with your downloaded models. By default the REST API to generate completions is available at `http://localhost:11434/api/generate`. The REST API exposes a number of additional endpoints for chat, managing models, and more. You can find more information about the REST API [here](https://github.com/ollama/ollama/blob/main/docs/api.md).
+
+To use the REST API, you make a POST request to the `/api/generate` endpoint with the following JSON payload:
+
+```json
+{
+  "prompt": "Your promt",
+  "model": "The model that you want to use",
+}
+```
+
+The `prompt` field is the text you want to generate a completion for. The `model` field is the name of the model you want to use. There are additional parameters you can pass in such as `stream`, `context`, `format`, and others, but the only required parameters are the `prompt` and `model` fields.
+
+Let's try it out! Let's make a POST request to the `/api/generate` endpoint with the following JSON payload:
+
+```json
+{
+  "prompt": "Write a Python function that takes a string and returns the number of vowels in the string.",
+  "model": "codellama",
+}
+```
+
+You can use a tool like [Postman](https://www.postman.com/) to make the request. Once you have the request set up, you can send it to the `/api/generate` endpoint and see the response.
+
+![Ollama from Postman](https://storage.googleapis.com/sourcegraph-assets/blog/local-code-completion-with-ollama-and-cody/ollama-postman.png)
+
+## Local code completion with Ollama and Cody
+
+Now that you have Ollama installed and running locally, you can use it with Cody to get local code completion. To do this, you need to install the [Cody VS Code extension](https://marketplace.visualstudio.com/items?itemName=sourcegraph.cody-ai). Once you have the extension installed, you can configure it to use Ollama by following these steps: 
```

Comment:
Since they know the terminal you could give them this option to install Cody:

`code --install-extension sourcegraph.cody-ai`